### ISSUE TITLE
Windows 11 registration sometimes fails

### ISSUE THREAD

**Issue by JeromeMartinez** [MediaArea/MediaInfo#993]
Up to now I received 2 reports about this issue. The second person provided this:
> MD5 hash for 64-bit only .exe: 5850DAEF7B2FD94A04D6A4001AE452FF  32/64 hash: 8644818749031AAE77B66462BFC1CD97. Size is 14,266,408 bytes, "size on disk" is 14,274,560 bytes.  All from your website.  Windows 11 Pro machine with Intel Core CPU, Windows 11 installed via Rufus bypass.  This is the first problem I've had installing any app in a very long time.   I have no explanation and v24.11.1 works fine.  Screenshot of message when running after install below.
![media](https://github.com/user-attachments/assets/8bbf53f4-3181-428e-a613-1d97ebc9a7ef)

I think we should at least have a fallback, not leaving the program if the registration fails.

For reference the first person only provided:
>  In my Windows 11 install v24.12 throws an 'unable to run, perhaps you don't have proper permissions' error  message.  Tried setting the .exe to run in admin mode, still no go.  Reverted to v24.11 and everything is fine again. 

Different message but not possible to have more information ("I do not have any interest in being your guinea pig, so I'll not be responding to any of your specific questions." answer to my request for more details after having said that he could be less pretentious in his first message 😅)

**Comment by cjee21** [comment]
**1st person**
I'm not sure if that error is even the exact error produced by Windows. The style does not sound like something Windows would show and searching for it produces no exact match. We cannot help when help is refused.

**2nd person**
MediaInfo should still run if package failed to install. It should be the same as running on Windows 10 as package is not installed on Windows 10. This error appearing actually means the package is already installed. Not sure what Windows means by could not be registered. Searching for this error shows cases where Windows built-in apps fail to launch with the same error so it may be issue with the particular Windows installation and not the app.

To get more info about what is happening with the package, the following PowerShell commands can be used:
- Check if package is provisioned (this step is done by `MediaInfo_PackageHelper`)
   ```powershell
   Get-AppxProvisionedPackage -Online | Where-Object {$_.DisplayName -match "MediaInfo"}
   ```
- Check if package is installed for current user (this step is done automatically by Windows)
   ```powershell
   Get-AppxPackage MediaInfo
   ```

Observation: The screenshot showing error dialog in dark colour with white titlebar and black OK text as well as mention of Rufus bypass hints that this user is the type that tweaks Windows likely beyond what is officially documented/supported by Microsoft which may or may not have anything to do with this issue.

**Comment by JeromeMartinez** [comment]
Answer from the user:

1.  When I click ok the box just disappears and that's it.

2. I could be missing it but with v24.12 installed I don't see anything regarding Mediainfo in the right click context menu.  Note that I have [StartAllBack](https://www.startallback.com/) running that provides the right click context menu in a single click (could possibly make some changes to it).

3. See results of of PowerShell commands run just below.

![media](https://github.com/user-attachments/assets/04926677-9cde-4a10-8e5f-919f9f17e0d0)

**Comment by cjee21** [comment]
Both provisioning and install/registration are fine as there is `MediaInfo` package returned for both commands with `Status: Ok` on last line.

I have no idea what StartAllBack actually does to Windows but an app with this name has been previously blocked by Microsoft/Windows.

**Comment by JeromeMartinez** [comment]
If we can not fix it, would it be at least a workaround? Which line of code could produce this error? We could either smoothly ignore the error rather than "disappearing" (I am not sure to understand if the box is disappearing or MediaInfo completely), or having a possibility to not launch the command (but if the Windows Explorer related checkbox is unchecked, which part of the code creates the issue?)

**Comment by cjee21** [comment]
It seems to be produced by Windows while attempting to start MediaInfo.exe. Based on other occurrences of similar error online, it can even happen while opening an image file with Windows built-in Photos app when the Windows installation is messed up.

![](https://filestore.community.support.microsoft.com/api/images/306b6dd3-43d0-46ba-bfb6-51e87f234994?upload=true&fud_access=wJJIheezUklbAN2ppeDns8cDNpYs3nCYjgitr%2BfFBh2dqlqMuW7np3F6Utp%2FKMltnRRYFtVjOMO5tpbpW9UyRAwvLeec5emAPixgq9ta07Dgnp2aq5eJbnfd%2FU3qhn54pSu5WUhfDMaxLh58Ra7mZdfYuZQjEixGaW6SA%2F5exKa0jysG9RJps%2FuvjvSUsq3tqvIX1yHA%2FI1IVSixGwO3QXPGE6dFOUojE5zB%2FEP058hbrr%2B0v8kWTf9poMRmlOGq0ZEt3Y%2FMnBCu%2Fti4XTLsM4jAFyzzfYzTup0woDA%2BZcYRIrJx8LqSAbAfvF6K63G1TF%2FaKmbs8ZFq3SeRRFmgqdt%2Bh0CVbLR4Q9v2SfLC0VjFoPetk4ZpDHRlkGZpAqN4lxKZ9ca35Pc7UBlsnIRidoKZWL8BvRBHozJKfmWIML8%3D)

As a workaround, the user can try to remove the MSIX package. Can be done by using PowerShell but will need to be done every time MediaInfo is updated unless the cause of the issue is rectified.

**Comment by JeromeMartinez** [comment]
>  Can be done by using PowerShell 

Please remind me the command?

> but will need to be done every time MediaInfo is updated unless the cause of the issue is rectified.

We could check if MediaInfo is already installed and the MSIX package not registered and we don't register it if it is the case.

**Comment by cjee21** [comment]
> We could check if MediaInfo is already installed and the MSIX package not registered and we don't register it if it is the case.

But it has to be installed for users updating from older versions or re-attempt to install if the previous installation somehow fails.

**Comment by JeromeMartinez** [comment]
> But it has to be installed for users updating from older versions or re-attempt to install if the previous installation somehow fails.

True. Let's first see with removing the MSIX package and we see if other people are impacted before doing something.

**Comment by cjee21** [comment]
> Please remind me the command?

[Remove-AppxProvisionedPackage](https://learn.microsoft.com/en-us/powershell/module/dism/remove-appxprovisionedpackage?view=windowsserver2025-ps) to remove the provisioned package
[Remove-AppxPackage](https://learn.microsoft.com/en-us/powershell/module/appx/remove-appxpackage?view=windowsserver2025-ps) to remove the package that has been installed by Windows for the current user

After running, both previous commands in posts above should return nothing if all successfully removed.

**Comment by cjee21** [comment]
> we see if other people are impacted before doing something.

If too many people having issues, we should just put the fully packaged VCL or Qt GUI in the Store and only do the new shell extensions there and direct Windows 11 users to the Store version.

**Comment by JeromeMartinez** [comment]
> only do the new shell extensions there and direct Windows 11 users to the Store version.

I hope that we won't do that... We'll see.

**Comment by JeromeMartinez** [comment]
> [Remove-AppxProvisionedPackage](https://learn.microsoft.com/en-us/powershell/module/dism/remove-appxprovisionedpackage?view=windowsserver2025-ps) to remove the provisioned package
[Remove-AppxPackage](https://learn.microsoft.com/en-us/powershell/module/appx/remove-appxpackage?view=windowsserver2025-ps) to remove the package that has been installed by Windows for the current user

Sorry to ask for more, but as I don't use to manage that I prefer to be sure, what are the expected exact commands? We user needs to do both?

**Comment by cjee21** [comment]
> what are the expected exact commands? 

```powershell
Remove-AppxProvisionedPackage -PackageName "MediaInfo_24.12.0.0_neutral__0z25jj8eabac0" -Online
```

```powershell
Remove-AppxPackage -Package "MediaInfo_24.12.0.0_neutral__0z25jj8eabac0"
```

The package name changes in each version as well as when the publisher name changes so the exact name as shown by the previous commands in https://github.com/MediaArea/MediaInfo/issues/993#issuecomment-2541686449 should be used.

> We user needs to do both?

Do both to ensure complete removal. After that repeat the commands in https://github.com/MediaArea/MediaInfo/issues/993#issuecomment-2541686449 which should return nothing.

**Comment by cjee21** [comment]
The real solution is likely to reinstall Windows: https://answers.microsoft.com/en-us/windows/forum/all/package-could-not-be-registered-windows-11/a725a62d-124f-4536-bf88-8fef32d3397e
